__tests__/pet.test.js/pet.test.js

```diff
@@ -0,0 +1,221 @@
+    expect(pet.age).toEqual(0);
+   });
+    
+});
+
+describe('growUp', () => {
+    it('increments the age by 1', () => {
+        const pet = new Pet('Fido');
+        pet.growUp();
+        expect(pet.age).toEqual(1);
+    });
+
+    it('throws an error if the pet is not alive', () => {
+        const pet = new Pet('Fido');
+  
+        pet.age = 30;
+  
+        expect(() => pet.growUp()).toThrow('Your pet is no longer alive :(');
+      });
+});
+
+describe('gettingUnhealthier', () => {
+    it('increases the hunger by 5 when age increases', () => {
+        const pet = new Pet('Fido');
+        pet.growUp();
+        expect(pet.hunger).toEqual(5);
+    });
+    it('decreases the fitness by 3 when age increases', () => {
+        const pet = new Pet('Fido');
+        pet.growUp();
+        expect(pet.fitness).toEqual(7);
+    });
+});
+
+describe('walk', () => {
+    it('increases fitness by 4 to a maximum of 10', () => {
+        const pet = new Pet('fido');
+        
+        pet.fitness = 8;
+        pet.walk();
+
+        expect(pet.fitness).toEqual(10);
+    });
+
+    it('throws an error if the pet is not alive', () => {
+        const pet = new Pet('Fido');
+  
+        pet.age = 30;
+  
+        expect(() => pet.walk()).toThrow('Your pet is no longer alive :(');
+      });
+});
+
+describe('feed', () => {
+
+    it('decreases hunger level to a minimum of 0', () => {
+        const pet = new Pet('fido');
+
+        pet.hunger = 3;
+        pet.feed();
+
+        expect(pet.hunger).toEqual(0);
+    });
+
+    it('throws an error if the pet is not alive', () => {
+        const pet = new Pet('Fido');
+  
+        pet.age = 30;
+  
+        expect(() => pet.feed()).toThrow('Your pet is no longer alive :(');
+      });
+});
+
+describe('checkUp', () => {
+    it('returns how the pet is feeling', () => {
+        const pet = new Pet('fido');
+
+        pet.hunger = 5;
+        pet.fitness = 3;
+        
+        expect(pet.checkUp()).toEqual('I am hungry AND I need a walk');
+
+    });
+
+    it('returns how the pet is feeling', () => {
+        const pet = new Pet('fido');
+
+        pet.hunger = 0;
+        pet.fitness = 4;
+        
+        expect(pet.checkUp()).toEqual('I feel great!');
+
+    });
+
+    it('returns how the pet is feeling', () => {
+        const pet = new Pet('fido');
+
+        pet.hunger = 0;
+        pet.fitness = 2;
+        
+        expect(pet.checkUp()).toEqual('I need a walk');
+
+    });
+
+    it('returns how the pet is feeling', () => {
+        const pet = new Pet('fido');
+
+        pet.hunger = 6;
+        pet.fitness = 5;
+        
+        expect(pet.checkUp()).toEqual('I am hungry');
+
+    });
+
+    it('throws an error if the pet is not alive', () => {
+        const pet = new Pet('Fido');
+  
+        pet.age = 30;
+  
+        expect(() => pet.checkUp()).toThrow('Your pet is no longer alive :(');
+      });
+
+
+});
+
+describe('isAlive', () => {
+    it('returns if the pet is alive', () => {
```

Comment:
These could be better named. This one could be 'returns true if hunger age and fitness all within allowed range', just as the next one could be 'returns false if the age is outside allowed range'